scripts/k8s_deploy_kubeflow.sh

```diff
@@ -111,35 +105,63 @@ function stand_up() {
   mkdir /tmp/kf-download
   cd /tmp/kf-download
   curl -O -L ${KFCTL_URL}
-  tar -xvf kfctl_${KUBEFLOW_TAG}_linux.tar.gz
+  tar -xvf ${KFCTL_FILE}
   mv kfctl ${KFCTL}
   popd
   rm -rf /tmp/kf-download
 
-  # Initialize and apply the Kubeflow project using the specified config
-  ${KFCTL} init ${KFAPP} --config=${CONFIG} -V
-  cd ${KFAPP}
-  ${KFCTL} generate all -V
-  ${KFCTL} apply all -V
+  # Create directory for KF files
+  mkdir ${KF_DIR}
 
-  echo "cd ${KFAPP} && ${KFCTL} delete -V k8s; cd && sudo rm -rf ${KFAPP}; sudo rm ${KFCTL}" > ${KUBEFLOW_DEL_SCRIPT}
-  echo "cd ${KFAPP} && ${KFCTL} delete -V all; cd && sudo rm -rf ${KFAPP}; sudo rm ${KFCTL}" > ${KUBEFLOW_DEL_SCRIPT}_full.sh
+  # Make cleanup scripts first in case deployment fails
+  # TODO: This kfctl delete seems to be failing due to a Kubeflow config bug
+  echo "cd ${KF_DIR} && ${KFCTL} delete -V -f ${CONFIG_FILE} --delete_storage; cd && sudo rm -rf ${KF_DIR}" > ${KUBEFLOW_DEL_SCRIPT}
   chmod +x ${KUBEFLOW_DEL_SCRIPT}
-  chmod +x ${KUBEFLOW_DEL_SCRIPT}_full.sh
+
+  # Initialize and apply the Kubeflow project using the specified config. We do this in two steps to allow a chance to customize the config
+  cd ${KF_DIR}
+  ${KFCTL} build -V -f ${CONFIG_URI}
+  # TODO: Add potential CONFIG customizations here in CONFIG_FILE
+  ${KFCTL} apply -V -f ${CONFIG_FILE}
+}
+
+
+# Modify the ns finalizers so they don't wait for async processes to complete
+function fix_terminating_ns() {
+  kubectl proxy &
+  for ns in ${@}; do
+    kubectl get namespace ${ns} -o json |jq '.spec = {"finalizers":[]}' > "/tmp/temp_${ns}.json"
+    curl -k -H "Content-Type: application/json" -X PUT --data-binary @"/tmp/temp_${ns}.json" 127.0.0.1:8001/api/v1/namespaces/${ns}/finalize
+  done
 }
 
 
 function tear_down() {
-  if [ ${KUBEFLOW_FULL_DELETE} ]; then
-    bash ${KUBEFLOW_DEL_SCRIPT}_full.sh
+  # Kubeflow use leads to some user created namespaces that are not torn down during kfctl delete
+  namespaces="kubeflow"
 
-    # Kubeflow use leads to some user created namespaces that are not torn down during kfctl delete
-    additional_namespaces="kubeflow-anonymous ${KUBEFLOW_EXTRA_NS}"
-    echo "Deleting additional namespaces ${additional_namespaces}, this may take several minutes"
-    kubectl delete ns ${additional_namespaces}
-  else
-    bash ${KUBEFLOW_DEL_SCRIPT}
+  # Delete other NS that were installed. These might be part of other apps and is slightly dangerous
+  if [ "${KUBEFLOW_FULL_DELETE}" == "true" ]; then
+    namespaces=" ${namespaces} kubeflow-anonymous auth anonymous cert-manager istio-system knative-serving ${KUBEFLOW_EXTRA_NS}"
```

Comment:
`kubeflow-anonymous`, `anonymous` don't exist anymore - and we should remove `kubeflow`